rezervo/providers/sats/provider.py

```diff
@@ -0,0 +1,311 @@
+import datetime
+import re
+from typing import Optional, Union
+
+from aiohttp import ClientSession, FormData
+
+from rezervo.consts import WEEKDAYS
+from rezervo.errors import AuthenticationError, BookingError
+from rezervo.http_client import HttpClient, create_client_session
+from rezervo.models import SessionState
+from rezervo.providers.provider import Provider
+from rezervo.providers.sats.auth import authenticate_session
+from rezervo.providers.sats.helpers import retrive_sats_page_props
+from rezervo.providers.sats.schema import (
+    SatsBookingsResponse,
+    SatsClass,
+    SatsDaySchedule,
+    SatsLocationIdentifier,
+    SatsScheduleResponse,
+    SatsWeekSchedule,
+)
+from rezervo.providers.sats.urls import (
+    BOOKING_URL,
+    BOOKINGS_PATH,
+    BOOKINGS_URL,
+    UNBOOK_URL,
+    schedule_url,
+)
+from rezervo.providers.schema import LocationIdentifier
+from rezervo.schemas.config.user import (
+    ChainUser,
+    ChainUserCredentials,
+    Class,
+    ClassTime,
+)
+from rezervo.schemas.schedule import (
+    RezervoActivity,
+    RezervoClass,
+    RezervoDay,
+    RezervoInstructor,
+    RezervoLocation,
+    RezervoSchedule,
+    SessionRezervoClass,
+    UserSession,
+)
+from rezervo.utils.category_utils import determine_activity_category
+
+
+class SatsProvider(Provider[ClientSession, SatsLocationIdentifier]):
+    async def _authenticate(
+        self, chain_user: ChainUser
+    ) -> Union[ClientSession, AuthenticationError]:
+        session = create_client_session()
+        return await authenticate_session(
+            session, chain_user.username, chain_user.password
+        )
+
+    async def find_class_by_id(
+        self, class_id: str
+    ) -> Union[RezervoClass, BookingError, AuthenticationError]:
+        provider_location_id_match = re.search(r"(\d+)p", class_id)
```

Comment:
Maybe create a helper function for the location id extraction from `class_id`?

Reply:
Implemented in https://github.com/mathiazom/rezervo/pull/43/commits/4ab071ef60185945e9c1fe7cd396e4174c4f9232